server/src/test/java/com/google/coffeehouse/common/PersonTest.java

```diff
@@ -36,69 +41,105 @@
   private static final String ALT_PRONOUNS = "New Pronouns";
   private static final String IDENTIFICATION_STRING = "predetermined-identification-string";
   private Person.Builder personBuilder;
+  private Map personInfo;
   
   @Mock private IdentifierGenerator idGen;
 
   @Before
   public void setUp() {
-    idGen = Mockito.mock(IdentifierGenerator.class);
-    Mockito.when(idGen.generateId()).thenReturn(IDENTIFICATION_STRING);
+    personInfo = new HashMap<String, String>();
+
+    idGen = mock(IdentifierGenerator.class);
+    when(idGen.generateId()).thenReturn(IDENTIFICATION_STRING);
 
     personBuilder = Person.newBuilder(EMAIL, NICKNAME);
   }
 
   @Test
-  public void getExistingNickname() {
+  public void getNickname_exists() {
     Person p = personBuilder.build();
     Assert.assertEquals(NICKNAME, p.getNickname());
   }
 
   @Test
-  public void getExistingEmail() {
+  public void getEmail_exists() {
     Person p = personBuilder.build();
     Assert.assertEquals(EMAIL, p.getEmail());
   }
 
   @Test
-  public void getExistingUserId() {
+  public void getUserId_exists() {
     Person p = personBuilder.setIdGenerator(idGen).build();
     Assert.assertEquals(IDENTIFICATION_STRING, p.getUserId());
   }
 
   @Test
-  public void getExistingPronouns() {
+  public void getPronouns_exists() {
     Person p = personBuilder.setPronouns(PRONOUNS).build();
     Assert.assertTrue(p.getPronouns().isPresent());
     Assert.assertEquals(PRONOUNS, p.getPronouns().get());
   }
 
   @Test
-  public void getNoPronouns() {
+  public void getPronouns_notExist() {
     Person p = personBuilder.build();
     Assert.assertFalse(p.getPronouns().isPresent());
   }
 
   @Test
-  public void setNewNickname() {
+  public void setNickname() {
     Person p = personBuilder.build();
     p.setNickname(ALT_NICKNAME);
     Assert.assertEquals(ALT_NICKNAME, p.getNickname());
   }
 
   @Test
-  public void setNewEmail() {
+  public void setEmail() {
     Person p = personBuilder.build();
     p.setEmail(ALT_EMAIL);
     Assert.assertEquals(ALT_EMAIL, p.getEmail());
   }
 
   @Test
-  public void setNewPronouns() {
+  public void setPronouns() {
     Person p = personBuilder.build();
     p.setPronouns(ALT_PRONOUNS);
     Assert.assertTrue(p.getPronouns().isPresent());
     Assert.assertEquals(ALT_PRONOUNS, p.getPronouns().get());
   }
 
+  @Test 
+  public void fromMap_invalidInput() {
+    personInfo.put(Person.NICKNAME_FIELD_NAME, NICKNAME);
+    Assert.assertThrows(IllegalArgumentException.class, new ThrowingRunnable() {
+        @Override
+        public void run() throws Throwable {
+          Person.fromMap(personInfo);
+        }
+    });
```

Comment:
This is Java 8 so you should be able to use a lambda function here.

```suggestion
    Assert.assertThrows(IllegalArgumentException.class, () -> {
        Person.fromMap(personInfo);
    });
```

Reply:
Good catch! This has been fixed and pushed.